src/pxfw.cpp

```diff
@@ -0,0 +1,654 @@
+		} else
+		if(!strncmp(dev->dev,"DVDR   PX-708A2",15)) {
+			dev->dev_ID=PLEXTOR_708A2;
+			*FWSZ=1028096;
+		} else
+		if(!strncmp(dev->dev,"DVDR   PX-712A",14)) {
+			dev->dev_ID=PLEXTOR_712;
+			*FWSZ=1028096;
+		} else
+		if(!strncmp(dev->dev,"DVDR   PX-714A",14)) {
+			dev->dev_ID=PLEXTOR_714;
+			*FWSZ=983040;
+		} else
+		if(!strncmp(dev->dev,"DVDR   PX-716A ",15)) {
+			dev->dev_ID=PLEXTOR_716;
+			*FWSZ=983040;
+		} else
+		if(!strncmp(dev->dev,"DVDR   PX-716AL",15)) {
+			dev->dev_ID=PLEXTOR_716AL;
+			*FWSZ=983040;
+		} else
+		if(!strncmp(dev->dev,"DVDR   PX-755A",14)) {
+			dev->dev_ID=PLEXTOR_755;
+			*FWSZ=2031616;
+		} else
+		if(!strncmp(dev->dev,"DVDR   PX-760A",14)) {
+			dev->dev_ID=PLEXTOR_760;
+			*FWSZ=2031616;
+		} else
+			dev->dev_ID=PLEXTOR_OLD;
+	}
+}
+
+/*
+int plextor_read_eeprom(drive_info* dev, unsigned char idx, unsigned int sz) {
+//	char*		data;
+	unsigned int	i,j;
+	int		offs=idx*sz;
+	unsigned char*  buf=dev->rd_buf+offs;
+
+	dev->cmd[0] = 0xF1;
+	dev->cmd[1] = 0x01;
+	dev->cmd[7] = idx;
+	dev->cmd[8] = (sz >> 8) & 0xFF;
+	dev->cmd[9] = sz & 0xFF;
+	if ((dev->err=dev->cmd.transport(READ,buf,sz) ))
+		{ sperror ("read EEPROM",dev->err); 	return (0); }
+
+	printf("EEPROM block #%d:\n",idx);
+	for(i=0;i<(sz/0x10);i++) {
+	    printf("| %X0 | ", i);
+	    for(j=0;j<0x10;j++) printf("%02X ",buf[i*0x10+j]);
+	    printf("|");
+	    for(j=0;j<0x10;j++) {
+		if (buf[i*0x10+j] > 0x20) printf("%c",buf[i*0x10+j]);
+		else printf(" ");
+	    }
+	    printf("|\n");
+	};
+	return 1;
+}
+*/
+
+int fwblk_send(drive_info* dev, int offs, int blksz, bool last) {
+	dev->cmd[0] = 0x3B;
+	dev->cmd[1] = last ? 0x05:0x04;
+	dev->cmd[2] = 0x00;
+	dev->cmd[3] = (offs >> 16) & 0xFF;
+	dev->cmd[4] = (offs >> 8) & 0xFF;
+	dev->cmd[5] = offs & 0xFF;
+	dev->cmd[6] = (blksz >> 16) & 0xFF;
+	dev->cmd[7] = (blksz >> 8) & 0xFF;
+	dev->cmd[8] = blksz & 0xFF;
+	dev->cmd[9] = 0x00;
+	dev->cmd[10]= 0x00;
+	dev->cmd[11]= 0x00;
+
+	if ((dev->err=dev->cmd.transport(WRITE,dev->rd_buf,blksz) )){
+		sperror ("SEND_FWBLK",dev->err);
+		if (!last) return 1;
+	}
+	return 0;
+}
+
+
+int fwblk_read(drive_info* dev, int offs, int blksz, unsigned char flag) {
+        printf("data read...\n");
+	return 1;
+}
+
+/*
+	dev->cmd[0] = 0x3B;
+//	cmd[1] = last ? 0x05:0x04;
+	dev->cmd[1] = flag;
+
+	dev->cmd[2] = 0x00;
+	dev->cmd[3] = (offs >> 16) & 0xFF;
+	dev->cmd[4] = (offs >> 8) & 0xFF;
+	dev->cmd[5] = offs & 0xFF;
+	dev->cmd[6] = (blksz >> 16) & 0xFF;
+	dev->cmd[7] = (blksz >> 8) & 0xFF;
+	dev->cmd[8] = blksz & 0xFF;
+	dev->cmd[9] = 0x00;
+	dev->cmd[10]= 0x00;
+	dev->cmd[11]= 0x00;
+
+	if ((dev->err=dev->cmd.transport(READ,dev->rd_buf,blksz) )){
+		sperror ("READ_FWBLK",dev->err);
+		return 1;
+	}
+	return 0;
+}
+*/
+
+void csum_init(unsigned short int *csum, int FW_dev_ID){
+    *csum = 0;
+    if (FW_dev_ID <= PLEXTOR_PREMIUM) *csum = 0x8000;
+}
+
+void csum_update(unsigned short int *csum, unsigned char* buf, int len) {
+	int		i;
+	unsigned int	ad;
```

Comment:
Codacy found an issue: [The scope of the variable 'ad' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)